lms/djangoapps/lms_xblock/mixin.py

```diff
@@ -150,11 +208,17 @@ def validate(self):
         validation = super(LmsBlockMixin, self).validate()
         has_invalid_user_partitions = False
         has_invalid_groups = False
+
         for user_partition_id, group_ids in self.group_access.iteritems():
             try:
                 user_partition = self._get_user_partition(user_partition_id)
             except NoSuchUserPartitionError:
                 has_invalid_user_partitions = True
+            except AttributeError:
+                try:
+                    user_partition = self.user_partitions[user_partition_id]
```

Comment:
@bbaker6225 if I'm reading the code correctly, you can rely on self._get_user_partition(user_partition_id) to raise a NoSuchUserPartitionError if a partition does not exist. What happens if you just don't catch AttributeError?